### ISSUE TITLE
Make Pack an incremental msbuild target

### ISSUE THREAD

**Issue by davkean** [NuGet/Home#4363]
1. File -> New Class Library (.NET Standard)
2. Right-click project, choose **Properties**
3. Click **Package**, check **Generate NuGet package on build**
4. Build
5. Wait 1 minute
6. Build

Expected: For the Nupkg not to be regenerated
Actual: Actual, the nupkg is being regenerated

You can see it rerunning the targets:

```
1>Target _GetPackageFiles:
1>Target GenerateNuspec:
1>  Task "ConvertToAbsolutePath"
1>  Done executing task "ConvertToAbsolutePath".
1>  Task "ConvertToAbsolutePath" skipped, due to false condition; ($(NuspecFile) != '') was evaluated as ( != '').
1>  Using "PackTask" task from assembly "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\NuGet.Build.Tasks.Pack\build\..\Desktop\NuGet.Build.Tasks.Pack.dll".
1>  Task "PackTask"
1>    Successfully created package 'bin\Debug\ClassLibrary12.1.0.0.nupkg'.
1>  Done executing task "PackTask".
1>Target Pack:
1>Target _PackAsBuildAfterTarget:
1>
1>Done building project "ClassLibrary12.csproj".
```

**Comment by rrelyea** [comment]
@rohit21agrawal / @davkean  - Need to understand importance -- what impact this will have.
ok to make in 4.0.1?

**Comment by rrelyea** [comment]
While this is very important, we can fix in 4.0.1.
